### PR TITLE
Fix TFViT

### PR DESCRIPTION
**Pull request by NielsRogge**
# What does this PR do?

Fix the code example of `TFViTForImageClassification` + fix the integration test.

cc @ydshieh

### CODE REVIEW

File: src/transformers/models/vit/modeling_tf_vit.py [hunk 1]
```diff
@@ -795,6 +795,7 @@ def call(
         Examples::
 
             >>> from transformers import ViTFeatureExtractor, TFViTForImageClassification
+            >>> import tensorflow as tf
```
Comment (sgugger) [inline src/transformers/models/vit/modeling_tf_vit.py:798]: It doesn't seem to be used in the example.

**Review by sgugger — APPROVED** [review]
Looks good, thanks for uploading the checkpoint!

**Comment by ydshieh** [comment]
LGTM, thanks!

File: src/transformers/models/vit/modeling_tf_vit.py (hunk 1 above)
Comment (sgugger) [reply src/transformers/models/vit/modeling_tf_vit.py:798]: Ah yes, I can see it. Ignore this comment then :-)

**Comment by ydshieh** [comment]
Hi, I found that there is something wrong (not introduced in this PR).

`model.save_pretrained("tmp", saved_model=True)` will give an error `AttributeError: 'ViTConfig' object has no attribute 'items'`.
(I found this when running `test_saved_model_creation_extended`)

@NielsRogge , would you mind to look this in this PR? Is `PretrainedConfig` supposed to have `items` attr?


Full example:
```
from transformers import ViTFeatureExtractor, TFViTForImageClassification
import tensorflow as tf
from PIL import Image
import requests
url = 'http://images.cocodataset.org/val2017/000000039769.jpg'
image = Image.open(requests.get(url, stream=True).raw)
feature_extractor = ViTFeatureExtractor.from_pretrained('google/vit-base-patch16-224')
model = TFViTForImageClassification.from_pretrained('google/vit-base-patch16-224')
inputs = feature_extractor(images=image, return_tensors="tf")
outputs = model(**inputs)
logits = outputs.logits
 # model predicts one of the 1000 ImageNet classes
predicted_class_idx = tf.math.argmax(logits, axis=-1)[0]
print("Predicted class:", model.config.id2label[int(predicted_class_idx)])

model.save_pretrained("tmp", saved_model=True)
```

**Pull request merged**